vcs/vcspath_test.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2015 The Serulian Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package vcs
+
+import (
+	"fmt"
+	"testing"
+)
+
+var _ = fmt.Printf
+
+func TestVCSParsing(t *testing.T) {
+	assertParses(t, "github.com/some/project", "github.com/some/project", "", "", "")
+	assertParses(t, "github.com/some/project:somebranch", "github.com/some/project", "somebranch", "", "")
+	assertParses(t, "github.com/some/project@sometag", "github.com/some/project", "", "sometag", "")
+	assertParses(t, "github.com/some/project//somesubdir", "github.com/some/project", "", "", "somesubdir")
+	assertParses(t, "github.com/some/project//somesubdir:somebranch", "github.com/some/project", "somebranch", "", "somesubdir")
+	assertParses(t, "github.com/some/project//somesubdir@sometag", "github.com/some/project", "", "sometag", "somesubdir")
+
+	assertParseError(t, "")
+	assertParseError(t, "github.com/foo/bar@@blah")
+	assertParseError(t, "github.com/foo/bar:@blah")
+	assertParseError(t, "github.com/foo/../bar")
+}
```

Comment:
Only if you're consistently going to have table driven tests for everything -- basically pick a test style and stick with it.